bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/CaddxBridgeHandler.java

```diff
@@ -0,0 +1,408 @@
+                logger.trace("Channel Type {} UID {}", c.getChannelTypeUID(), c.getUID());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("Initializing the Bridge handler.");
+
+        init();
+    }
+
+    @Override
+    public void dispose() {
+        CaddxCommunicator comm = communicator;
+        if (comm != null) {
+            comm.stop();
+            comm = null;
+        }
+
+        if (discoveryService != null) {
+            unregisterDiscoveryService();
+        }
+
+        super.dispose();
+    }
+
+    public @Nullable Thing findThing(CaddxThingType caddxThingType, @Nullable Integer partition, @Nullable Integer zone,
+            @Nullable Integer keypad) {
+        List<Thing> things = getThing().getThings();
+
+        Thing thing = null;
+
+        for (Thing t : things) {
+            try {
+                Configuration config = t.getConfiguration();
+                CaddxBaseThingHandler handler = (CaddxBaseThingHandler) t.getHandler();
+
+                if (handler != null) {
+                    CaddxThingType handlerCaddxThingType = handler.getCaddxThingType();
+
+                    if (handlerCaddxThingType.equals(caddxThingType)) {
+                        switch (handlerCaddxThingType) {
+                            case PANEL:
+                                thing = t;
+                                return thing;
+                            case KEYPAD:
+                                BigDecimal keypadAddress = (BigDecimal) config
+                                        .get(CaddxKeypadConfiguration.KEYPAD_ADDRESS);
+                                if (keypad == keypadAddress.intValue()) {
+                                    thing = t;
+                                    return thing;
+                                }
+                                break;
+                            case PARTITION:
+                                BigDecimal partitionNumber = (BigDecimal) config
+                                        .get(CaddxPartitionConfiguration.PARTITION_NUMBER);
+                                if (partition == partitionNumber.intValue()) {
+                                    thing = t;
+                                    return thing;
+                                }
+                                break;
+                            case ZONE:
+                                BigDecimal zoneNumber = (BigDecimal) config.get(CaddxZoneConfiguration.ZONE_NUMBER);
+                                if (zone == zoneNumber.intValue()) {
+                                    thing = t;
+                                    return thing;
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+
+                }
+            } catch (Exception e) {
```

Comment:
What would throw this?

Reply:
That is also code copied from the dscalarm binding.
I suppose you can arrive at this Exception only with a ClassCastException if the configuration is not correct. 
Should i change Exception to ClassCastException or should i delete the exception handling?

Reply:
I think removing the exception handling here would be better.

Reply:
Done